be/src/exec/json_scanner.cpp

```diff
@@ -841,4 +853,14 @@ Status JsonReader::_construct_column(simdjson::ondemand::value& value, Column* c
     return add_adaptive_nullable_column(column, type_desc, col_name, &value, !_strict_mode);
 }
 
+void JsonReader::_append_error_msg(std::string&& row, const std::string& error_msg) {
+    std::string row_with_meta;
+    if (_file_stream_buffer == nullptr || _file_stream_buffer->meta()->type() == ByteBufferMetaType::NONE) {
+        row_with_meta = std::move(row);
+    } else {
+        row_with_meta = fmt::format("{} [meta: {}]", row, _file_stream_buffer->meta()->to_string());
```

Comment:
It's better to process `KAFKA` first, then default for all other invalid or NONE situations.